File: py/examples/counter_unicast.py
# Mode / Unicast
# Launch the server in #unicast #mode and use `q.client` to manage client-local state.
# ---
from h2o_wave import main, app, Q, ui


@app('/demo')
async def serve(q: Q):
    if not q.client.initialized:
        q.client.count = 0
        q.page['example'] = ui.form_card(box='1 1 2 1', items=[
            ui.button(name='increment', label=f'Count={q.client.count}')
        ])
        q.client.initialized = True

    if q.args.increment:
        q.client.count += 1
        q.page['example'].increment.label = f'Count={q.client.count}'

    await q.page.save()

File: r/R/examples/counter_unicast.R
library(h2owave)
serve <- function(qo)
{
    qo$client$test <- ifelse(is.null(qo$client$test),0, qo$client$test)
    ifelse(qo$args$increment == TRUE,qo$client$test <- qo$client$test + 1,qo$client$test <- 0)
    if(length(qo$args) == 0 || qo$args$increment == FALSE){
     qo$page$add_card("button_entry",ui_form_card(
                                                box='1 1 12 10'
                                                ,items = list(
                                                              ui_button(
                                                                        name='increment'
                                                                        ,label=paste0('Count= ',qo$client$test)
                                                                        ,primary = TRUE
                                                              )
                                                )
                                                ))
    }
    else {
        qo$page$set("button_entry","items","0","button","label",paste0("Count=",qo$client$test))
    }
    qo$page$save()
}
app("/")
